pkg/sessions/tests/session_store_tests.go

```diff
@@ -0,0 +1,443 @@
+}
+
+// sessionStoreFunc is used in testInput to wrap the SessionStore interface.
+type sessionStoreFunc func() sessionsapi.SessionStore
+
+// PersistentStoreFastForwardFunc is used to adjust the time of the persistent
+// store to fast forward expiry of sessions.
+type PersistentStoreFastForwardFunc func(time.Duration) error
+
+// NewSessionStoreFunc allows any session store implementation to configure their
+// own session store before each test.
+type NewSessionStoreFunc func(sessionOpts *options.SessionOptions, cookieOpts *options.CookieOptions) (sessionsapi.SessionStore, error)
+
+func RunSessionStoreTests(newSS NewSessionStoreFunc, persistentFastForward PersistentStoreFastForwardFunc) {
+	Describe("Session Store Suite", func() {
+		var opts *options.SessionOptions
+		var ss sessionsapi.SessionStore
+		var input testInput
+
+		getSessionStore := func() sessionsapi.SessionStore {
+			return ss
+		}
+
+		BeforeEach(func() {
+			ss = nil
+			opts = &options.SessionOptions{}
+
+			// Set default options in CookieOptions
+			cookieOpts := &options.CookieOptions{
+				Name:     "_oauth2_proxy",
+				Path:     "/",
+				Expire:   time.Duration(168) * time.Hour,
+				Refresh:  time.Duration(1) * time.Hour,
+				Secure:   true,
+				HTTPOnly: true,
+				SameSite: "",
+			}
+
+			expires := time.Now().Add(1 * time.Hour)
+			session := &sessionsapi.SessionState{
+				AccessToken:  "AccessToken",
+				IDToken:      "IDToken",
+				ExpiresOn:    &expires,
+				RefreshToken: "RefreshToken",
+				Email:        "john.doe@example.com",
+				User:         "john.doe",
+			}
+
+			request := httptest.NewRequest("GET", "http://example.com/", nil)
+			response := httptest.NewRecorder()
+
+			input = testInput{
+				cookieOpts:            cookieOpts,
+				ss:                    getSessionStore,
+				session:               session,
+				request:               request,
+				response:              response,
+				persistentFastForward: persistentFastForward,
+			}
+		})
+
+		Context("with default options", func() {
+			BeforeEach(func() {
+				var err error
+				ss, err = newSS(opts, input.cookieOpts)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			SessionStoreInterfaceTests(&input)
+			if persistentFastForward != nil {
+				PersistentSessionStoreInterfaceTests(&input)
+			}
+		})
+
+		Context("with non-default options", func() {
+			BeforeEach(func() {
+				input.cookieOpts = &options.CookieOptions{
+					Name:     "_cookie_name",
+					Path:     "/path",
+					Expire:   time.Duration(72) * time.Hour,
+					Refresh:  time.Duration(2) * time.Hour,
+					Secure:   false,
+					HTTPOnly: false,
+					Domains:  []string{"example.com"},
+					SameSite: "strict",
+				}
+
+				var err error
+				ss, err = newSS(opts, input.cookieOpts)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			SessionStoreInterfaceTests(&input)
+			if persistentFastForward != nil {
+				PersistentSessionStoreInterfaceTests(&input)
+			}
+		})
+
+		Context("with a cipher", func() {
```

Comment:
Same as my other note: should we have a cipher present in all test cases now since validation requires a cipher to exist now?

I think I might've taken care of this in #632 if you want to punt in this PR.